pkg/cmd/server/start/start_master.go

```diff
@@ -100,21 +101,18 @@ func NewCommandStartMaster() (*cobra.Command, *MasterOptions) {
 		},
 	}
 
+	options.MasterArgs = NewDefaultMasterArgs()
+
 	flags := cmd.Flags()
 
-	flags.BoolVar(&options.WriteConfigOnly, "write-config", false, "Indicates that the command should build the configuration from command-line arguments, write it to the location specified by --config, and exit.")
-	flags.StringVar(&options.ConfigFile, "config", "", "Location of the master configuration file to run from, or write to (when used with --write-config). When running from a configuration file, all other command-line arguments are ignored.")
-
-	options.MasterArgs = NewDefaultMasterArgs()
-	// make sure that KubeConnectionArgs and MasterArgs use the same CertArgs for this command
-	options.MasterArgs.KubeConnectionArgs.CertArgs = options.MasterArgs.CertArgs
+	flags.Var(options.MasterArgs.ConfigDir, "write-config", "Indicates that the command should build the configuration from command-line arguments, write it to the location specified, and exit.")
```

Comment:
s/location specified/specified directory/ to make it clear a directory should be targeted